cap/one_for_one.go

```diff
@@ -42,40 +38,38 @@ func oneForOneRestartLoop(
 	supCtx context.Context,
 	eventNotifier EventNotifier,
 	supRuntimeName string,
+	supTolerance *errToleranceManager,
 	supChildren map[string]c.Child,
 	supNotifyCh chan<- c.ChildNotification,
-	wasComplete bool,
 	prevCh c.Child,
 	prevErr error,
-) *c.ErrorToleranceReached {
+) *ErrorToleranceReached {
 	for {
+		if prevErr != nil {
```

Comment:
The overloading of the `prevErr` parameter and the mutation in the loop is a little convoluted. The `errToleranceManager` could keep track of the last error?

Reply:
Good idea, let me work with that and see how it feels

Reply:
Uhm, the `prevErr` is a misnomer, I should call it `prevChErr` instead